src/c-writer.cc

```diff
@@ -3109,6 +3112,11 @@ void CWriter::WriteVarsByType(const Vars& vars,
       if (typeoffunc(var) == type) {
         if (count == 0) {
           Write(type, " ");
+          if (setjmp_safe) {
+            PushFuncSection("exceptions");
+            Write("volatile ");
+            PushFuncSection();
```

Comment:
I assume this means we only output this volatile when exceptions are enabled?

Reply:
yep!

Reply:
also, only when a try block is used.